ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateRequest.java

```diff
@@ -0,0 +1,176 @@
+    }
+  }
+
+  /**
+   * @param correlationId the correlation id for the request
+   * @param clientId the id of the client generating the request
+   * @param blobId the blob ID whose TTL needs to be updated
+   * @param expiresAtMs the absolute time (in ms) at which the blob must expire
+   * @param operationTimeMs the time of the operation (in ms)
+   */
+  public TtlUpdateRequest(int correlationId, String clientId, BlobId blobId, long expiresAtMs, long operationTimeMs) {
+    this(correlationId, clientId, blobId, expiresAtMs, operationTimeMs, CURRENT_VERSION);
+  }
+
+  /**
+   * @param correlationId the correlation id for the request
+   * @param clientId the id of the client generating the request
+   * @param blobId the blob ID whose TTL needs to be updated
+   * @param expiresAtMs the absolute time (in ms) at which the blob must expire
+   * @param operationTimeMs the time of the operation (in ms)
+   * @param version the version of the TtlUpdateRequest
+   */
+  TtlUpdateRequest(int correlationId, String clientId, BlobId blobId, long expiresAtMs, long operationTimeMs,
+      short version) {
+    super(TtlUpdateRequest, version, correlationId, clientId);
+    this.version = version;
+    this.blobId = blobId;
+    this.expiresAtMs = expiresAtMs;
+    this.operationTimeInMs = operationTimeMs;
+    sizeSent = 0;
+  }
+
+  @Override
+  public long writeTo(WritableByteChannel channel) throws IOException {
+    long written = 0;
+    if (bufferToSend == null) {
+      bufferToSend = ByteBuffer.allocate((int) sizeInBytes());
+      writeHeader();
+      bufferToSend.put(blobId.toBytes());
+      bufferToSend.putLong(expiresAtMs);
+      bufferToSend.putLong(operationTimeInMs);
+      bufferToSend.flip();
+    }
+    if (bufferToSend.remaining() > 0) {
+      written = channel.write(bufferToSend);
+      sizeSent += written;
+    }
+    return written;
+  }
+
+  @Override
+  public boolean isSendComplete() {
+    return sizeSent == sizeInBytes();
+  }
+
+  /**
+   * @return the blob ID whose TTL needs to be updated.
+   */
+  public BlobId getBlobId() {
+    return blobId;
+  }
+
+  /**
+   * @return the id of the account that the blob belongs to
+   */
+  public short getAccountId() {
+    return blobId.getAccountId();
+  }
+
+  /**
+   * @return the id of the container that the blob belongs to
+   */
+  public short getContainerId() {
+    return blobId.getContainerId();
+  }
+
+  /**
+   * @return the absolute time (in ms) at which the blob must expire
+   */
+  public long getExpiresAtMs() {
+    return expiresAtMs;
+  }
+
+  /**
+   * @return the time of the operation (in ms)
+   */
+  public long getOperationTimeInMs() {
+    return operationTimeInMs;
+  }
+
+  @Override
+  public long sizeInBytes() {
+    // header + blobId
+    // expires time ms
+    // op time ms
+    return super.sizeInBytes() + blobId.sizeInBytes() + 2 * Long.BYTES;
+  }
+
+  @Override
+  public String toString() {
```

Comment:
Looks like unit test doesn't cover `toString()` method.

Reply:
Yeah - we usually don't have tests for `toString()` (none of the other classes do either)